usvg/src/options.rs

```diff
@@ -132,11 +137,16 @@ pub struct Options {
     /// Default: empty
     #[cfg(feature = "text")]
     pub fontdb: fontdb::Database,
+
+    /// Specifies the way to parse `<image>` elements `xlink:href` value.
+    ///
+    /// Default: either parse `href` as DataUrl, or parse it as a path to the local image file.
+    pub image_href_resolver: Option<ImageHrefResolver<'a>>,
```

Comment:
I don't think resolver should be optional. It should be set to default by default. And default resolver creation should be cheap.

Reply:
The reason I did it is that:

>Current resvg resolver logic requires Options itself. So if we use `custom_href_resolver` as `Options` field we need to capture `Options` itself and it cannot be referenced, has to be owned Options.

Since existing default resolver logic requires `Options` (because inside the resolver there is SVG case where we recursively call render), I can't create a default resolver without having `Options` struct first. That's why `use_default_href_resolver` function exists as a separate step. First we create Options with resolver as None, then use them to define resolver itself.

So far don't see another way around it...